tests/test_ensemble.py

```diff
@@ -0,0 +1,381 @@
+"""Unit tests for ensemble module."""
+import tempfile
+import unittest
+
+import numpy as np
+
+import relentless
+
+class test_RDF(unittest.TestCase):
+    """Unit tests for relentless.RDF"""
+
+    def test_init(self):
+        """Test creation from data"""
+        #test valid construction
+        r = [1,2,3]
+        g = [2,9,5]
+        rdf = relentless.RDF(r=r, g=g)
+        np.testing.assert_allclose(rdf.table, np.array([[1,2],
+                                                        [2,9],
+                                                        [3,5]]))
+
+        #test invalid construction with r and g having different lengths
+        r = [1,2,3,4]
+        with self.assertRaises(ValueError):
+            rdf = relentless.RDF(r=r, g=g)
+
+        #test invalid construction with non-strictly-increasing r
+        r = [1,3,2]
+        with self.assertRaises(ValueError):
+            rdf = relentless.RDF(r=r, g=g)
+
+class test_Parallelepiped(unittest.TestCase):
+    """Unit tests for relentless.Parallelepiped"""
+
+    def test_init(self):
+        """Test creation from data."""
+        #test valid construction
+        p = relentless.Parallelepiped(a=(1,2,1),b=(3,4,5),c=(9,9,0))
+        np.testing.assert_allclose(p.a, np.array([1,2,1]))
+        np.testing.assert_allclose(p.b, np.array([3,4,5]))
+        np.testing.assert_allclose(p.c, np.array([9,9,0]))
+        np.testing.assert_allclose(p.matrix, np.array([[1,3,9],
+                                                       [2,4,9],
+                                                       [1,5,0]]))
+        self.assertAlmostEqual(p.volume, 36)
+
+        #test invalid construction
+        with self.assertRaises(TypeError):
+            p = relentless.Parallelepiped(a=(1,2,1),b=(3,4,5),c=(9,9))
```

Comment:
Add a check that construction fails if the vectors are not righthanded, i.e., the volume would be negative.